.github/workflows/functional-test.yml

```diff
@@ -44,23 +44,38 @@ jobs:
       TEST_PASS_THRESHOLD: 10
     # No hardware acceleration is available for emulators on Ubuntu:
     # https://github.com/marketplace/actions/android-emulator-runner#can-i-use-this-action-on-linux-vms
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v3
     - uses: actions/setup-node@v3
       with:
         node-version: lts/*
         check-latest: true
+    - run: curl -sSL https://bit.ly/install-xq | sudo bash
     - run: |
-        brew install xq
         npm install mocha-multi-reporters --save-dev
         npm install -g appium
         npm install --no-save mjpeg-consumer
       name: Install dev dependencies
     - uses: actions/setup-java@v3
       with:
         distribution: 'temurin'
-        java-version: '11'
+        java-version: '17'
+    - name: Enable KVM group perms
+      run: |
+        echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+        sudo udevadm control --reload-rules
+        sudo udevadm trigger --name-match=kvm
+    - name: Setup Android SDK
+      uses: android-actions/setup-android@v3
+    - name: AVD cache
```

Comment:
Caching actually needs two steps. Check the last example at https://github.com/ReactiveCircus/android-emulator-runner?tab=readme-ov-file#usage--examples